test/examples_test.go

```diff
@@ -81,6 +82,10 @@ func substituteEnv(input []byte, namespace string) ([]byte, error) {
 	// the test namespace using `ko create -n`
 	output = defaultNamespaceRE.ReplaceAll(output, []byte("namespace: "+namespace))
 
+	// Replace image names to arch specific ones, where it's necessary
+	for existingImage, archSpecificImage := range imagesMappingRE {
+		output = existingImage.ReplaceAll(output, archSpecificImage)
```

Comment:
I'm getting a little nervous about proliferating regexp-based replacements in our YAML files, it seems like it might end up making the tests brittle. Should we investigate using a purpose-built tool like kustomize to apply targetted replacements instead? (doesn't have to block this PR, just curious about your thoughts)

Reply:
I agree that massive regexp replacements might be dangerous. To support given regexp usage, it's for non amd64 only (so won't cover most of the test runs), it's for very specific case ("image: <name>") and regexp is already used in the same func.
Anyway, I'd like to spend some time to investigate kustomize usage option for examples tests(outside of this PR if you don't mind), because in general I agree with you that every work should be done by designed tool.

Reply:
sgtm, no need to do it in this PR but I'd like to consider it in future changes is all.